core/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -101,6 +101,11 @@
     public static final Setting<Boolean> INDEX_QUERY_CACHE_EVERYTHING_SETTING =
         Setting.boolSetting("index.queries.cache.everything", false, Property.IndexScope);
 
+    // This setting is an escape hatch in case not caching term queries would slow some users down
+    // Do not document.
+    public static final Setting<Boolean> INDEX_QUERY_CACHE_TERM_QUERIES_SETTING =
+        Setting.boolSetting("index.queries.cache.term_queries", true, Property.IndexScope);
```

Comment:
It seems like it would make more sense if the setting being `false` meant that term queries were not cached. Right now just looking at the name, it seems like setting it to `true` would mean that term queries  **are** cached.